packages/edit-post/src/components/options-modal/meta-boxes-section.js

```diff
@@ -25,7 +26,11 @@ export function MetaBoxesSection( { areCustomFieldsRegistered, metaBoxes, ...sec
 
 	return (
 		<Section { ...sectionProps }>
-			{ areCustomFieldsRegistered && <EnableCustomFieldsOption label={ __( 'Custom Fields' ) } /> }
+			{ areCustomFieldsRegistered && (
+				<PostTypeSupportCheck supportKeys="custom-fields">
```

Comment:
This check shouldn't be necessary because `getEditorSettings().enableCustomFields` should be `undefined` when the current post type doesn't support custom fields. 

It looks like this works correctly in WordPress 5.0 RC2 but not in WordPress 4.9 + Gutenberg 4.6.1.

I think that we should remove this check and make `enableCustomFields` work the same as it does in core.

Alternatively, we could remove this check and then simply wait for the Gutenberg plugin to require WordPress 5.0 as its minimum supported version.

Reply:
I agree about
> I think that we should remove this check and make enableCustomFields work the same as it does in core.

That's what I did in https://github.com/WordPress/gutenberg/pull/12394/commits/8a3b3f04164fb3968fc557afd03f1210a92b7fc0